### PR TITLE
Import grafana dashboards and datasources from cluster

### LINKED ISSUE #5368: FEATURE REQUEST: Add dashboards from other charts to grafana
FEATURE REQUEST
We are currently deploying several services in one cluster using helm. Each of those services offers his own dashboard/datasource. We are looking for a simple way to import them into a central grafana instance(preferably using the 5.0 feature of provisioning). Other solutions like using curl to import or using tools like [wizzy](https://github.com/utkarshcmu/wizzy) feel unhandy. Also dashboards/datasources should be removed once the connected service is deleted.

### PR DESCRIPTION
**Pull request by pulledtim**
**What this PR does / why we need it**:
Offers the possibility to collect configmaps from within the cluster and add the there stored dashboards/datasources to grafana. Adds a new service account with the needed permissions.

Details: Depending on the values, up to two sidecars are deployed in the pod. Those containers access the kubernetes api and watch on config map changes. Those are filtered for configmaps with a defined value. The files in those configmaps are stored in a shared folder and imported by grafana. When a configmap is removed, the connected file is removed too (and grafana removes it too)
**Which issue this PR fixes** 
fixes  #5368

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by pulledtim** [comment]
@zanhsieh @rtluckie

File: stable/grafana/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: grafana
-version: 1.2.2
+version: 1.2.3
```
Comment (timstoop) [inline stable/grafana/Chart.yaml]: As you're adding a new feature, a minor version bump is needed.

**Review by timstoop — CHANGES_REQUESTED** [review]
Please make the ServiceAccount and rbac optional, as described in the Best Practises: https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

File: stable/grafana/templates/deployment.yaml [hunk 2]
```diff
@@ -48,6 +51,36 @@ spec:
...
+          env:
+            - name: LABEL
+              value: "{{ .Values.sidecar.datasources.label }}"
+            - name: FOLDER
+              value: /etc/grafana/provisioning/datasources
+          resources:
+{{ toYaml .Values.sidecar.resources | indent 12 }}
+          volumeMounts:
+            - name: sc-datasources-volume
+              mountPath: /etc/grafana/provisioning/datasources
```
Comment (zanhsieh) [inline stable/grafana/templates/deployment.yaml]: mountPath needs to be consistent with line 118.

File: stable/grafana/values.yaml [hunk 3]
```diff
@@ -146,9 +146,30 @@ grafana.ini:
...
     check_for_updates: true
   log:
     mode: console
   grafana_net:
     url: https://grafana.net
+
+## Sidecars that collect the configmaps with specified label and stores the included files them into the respective folders
+## Requires at least Grafana 5 to work and can't be used together with parameters dashboardProviders, datasources and dashboards
+sidecar:
+  image: kiwigrid/k8s-sidecar:0.0.3
```
Comment (zanhsieh) [inline stable/grafana/values.yaml]: Suggest sidecar image should be provided on values.yaml as default. Please don't hardcode it.

**Comment by pulledtim** [comment]
Thank you @zanhsieh and @timstoop for your input. I worked on your remarks and hope to have covered all best practices now.

**Review by zanhsieh — APPROVED** [review]

**Comment by unguiculus** [comment]
Please follow our RBAC best practices:
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

**Comment by pulledtim** [comment]
@unguiculus Thank you for your remark. I changed the the rbac setting to be enabled by default, as described by the best practices. Sorry for missing that point. Hope I covered it completely now.

**Comment by pulledtim** [comment]
@unguiculus I adapted the PR to the current state of the grafana chart. Hope you can give me feedback again. Thank you for your time and patience

File: stable/grafana/values.yaml [hunk 4]
```diff
@@ -214,3 +215,23 @@ smtp:
   # `existingSecret` is a reference to an existing secret containing the smtp configuration
   # for Grafana in keys `user` and `password`.
   existingSecret: ""
+
+## Sidecars that collect the configmaps with specified label and stores the included files them into the respective folders
+## Requires at least Grafana 5 to work and can't be used together with parameters dashboardProviders, datasources and dashboards
+sidecar:
+  image: kiwigrid/k8s-sidecar:0.0.3
+  imagePullPolicy: Always
```
Comment (unguiculus) [inline stable/grafana/values.yaml]: IfNotPresent

File: stable/grafana/values.yaml [hunk 5]
```diff
@@ -214,3 +215,23 @@ smtp:
...
+  resources:
+    # limits:
+    #   cpu: 100m
+    #   memory: 100Mi
+    requests:
```
Comment (unguiculus) [inline stable/grafana/values.yaml]: Don't specify defaults for resources. Commeent them out.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/grafana/values.yaml (hunk 5 above)
Comment (monotek) [reply stable/grafana/values.yaml]: May i ask why?

**Comment by pulledtim** [comment]
Thank you @unguiculus for your remarks. I changed it according to your requests.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**